### PR TITLE
refactor tendermint-rs reexports

### LINKED ISSUE #3248: Only re-export current CometBFT version in use
In `namada_core`, we should only re-export `v0.37.x` related `tendermint-rs` modules, to avoid accidentally using `v0.34.x` or `v0.38.x` data structures, which might be incompatible.

### PR DESCRIPTION
**Pull request by tzemanovic**
## Describe your changes

Depends-On: #3670

- closes #3248 
- diff https://github.com/anoma/namada/compare/tomas/crate-refactors...tomas/refactor-tm-reexports

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes

### CODE REVIEW

File: crates/core/src/lib.rs [hunk 1]
```diff
@@ -23,8 +23,23 @@ pub mod bytes;
...
+        pub use tendermint::abci::{
+            types, Code, Event, EventAttribute, MethodKind,
+        };
+        pub use tendermint::v0_37::abci::*;
+    }
+    pub use tendermint::*;
+}
+/// Re-export of tendermint-proto v0.37
+pub mod tendermint_proto {
+    pub use tendermint_proto::google; // 💩
```
Comment (Fraccaman) [inline crates/core/src/lib.rs:40]: 💩💩💩

**Comment by tzemanovic** [comment]
@mergifyio refresh

**Comment by tzemanovic** [comment]
@mergifyio refresh

**Review by sug0 — APPROVED** [review]

**Comment by Fraccaman** [comment]
@mergify refresh

**Pull request merged**